llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -6411,6 +6411,40 @@ static Value *simplifyBinaryIntrinsic(Function *F, Value *Op0, Value *Op1,
       return Constant::getNullValue(ReturnType);
     break;
   }
+  case Intrinsic::ptrmask: {
+    if (isa<PoisonValue>(Op0) || isa<PoisonValue>(Op1))
+      return PoisonValue::get(Op0->getType());
+
+    // NOTE: We can't apply this simplifications based on the value of Op1
+    // because we need to preserve provenance.
+    if (Q.isUndefValue(Op0) || match(Op0, m_Zero()))
+      return Constant::getNullValue(Op0->getType());
+
+    if (Op1->getType()->getScalarSizeInBits() ==
```

Comment:
I believe this branch can be replaced with assert(Op1->getType()->getScalarSizeInBits() == Q.DL.getPointerIndexSizeInBits(Op0->getType()) since the verified now ensures the argument matches the index size since #69343

Reply:
I think it needs to be pointer size (as opposed to index size) as we are checking for useless masks. I.e if pointer size > index size all 1s mask isn't redundant.

Reply:
It is redundant since it does not affect more than the index bits (see clarification in the langref added in https://github.com/llvm/llvm-project/pull/69343)